File: ToughDashApr19/three.c
#include<stdio.h>

int main() {
    unsigned int num1, num2, t;

    scanf("%u", &t);

    for(unsigned int i=0;i<t;i++) {
        scanf("%u %u", &num1, &num2);

        num1 = (num1+num2)/2;

        if(num1 % 2 == 0)
            printf("Sadia will be happy.\n");
        else
            printf("Oops!\n");
    }
    return 0;
}

File: ToughDashApr19/three.py
t = int(input())

for x in range(t):
    line = input()
    num1, num2 = [int(x) for x in line.split()] # convert to INT in one line
    average = (num1 + num2) // 2 # Integer division

    if average % 2 == 0:
        print("Sadia will be happy.")
    else:
        print("Oops!") 